src/osd/ECBackend.cc

```diff
@@ -1556,30 +1567,52 @@ int ECBackend::get_min_avail_to_read_shards(
 int ECBackend::get_remaining_shards(
   const hobject_t &hoid,
   const set<int> &avail,
+  const set<int> &want,
```

Comment:
NIT: To make the code self-documenting it be nicer if any "int" that really is a shard be changed to a shard_id_t.